pkg/execute/executor.go

```diff
@@ -83,33 +76,30 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
 	empty := interactive.Message{}
 	rawCmd := format.RemoveHyperlinks(e.message)
 	rawCmd = strings.NewReplacer(`“`, `"`, `”`, `"`, `‘`, `"`, `’`, `"`).Replace(rawCmd)
-	clusterName := e.cfg.Settings.ClusterName
-	inClusterName := getClusterNameFromKubectlCmd(rawCmd)
-	botName := e.notifierHandler.BotName()
 	cmdCtx := CommandContext{
-		ClusterName:     clusterName,
-		CommGroupName:   e.commGroupName,
-		BotName:         botName,
+		ClusterName:     e.cfg.Settings.ClusterName,
 		RawCmd:          rawCmd,
+		CommGroupName:   e.commGroupName,
+		BotName:         e.notifierHandler.BotName(),
 		User:            e.user,
 		Conversation:    e.conversation,
 		Platform:        e.platform,
 		NotifierHandler: e.notifierHandler,
 		Mapping:         e.cmdsMapping,
 	}
-	execFilter, err := extractExecutorFilter(rawCmd)
+
+	flags, err := ParseBotkubeFlags(rawCmd)
 	if err != nil {
+		e.log.Errorf("while parsing command flags %q: %s", rawCmd, err.Error())
 		return respond(err.Error(), cmdCtx)
 	}
-	cmdCtx.ExecutorFilter = execFilter
 
-	args, err := shellwords.Parse(strings.TrimSpace(execFilter.FilteredCommand()))
-	if err != nil {
-		e.log.Errorf("while parsing command %q: %s", execFilter.FilteredCommand(), err.Error())
-		return respond("Cannot parse command. Please use 'help' to see supported commands.", cmdCtx)
-	}
+	cmdCtx.CleanCmd = flags.CleanCmd
+	cmdCtx.ClusterName = flags.ClusterName
```

Comment:
I spotted a bug:

```suggestion
	cmdCtx.ProvidedClusterName = flags.ClusterName
```

Correct? Please test your PR by yourself before pinging me next time as I'd like to finally test an accept it 🙂